Automattic/wp-calypso
# Gravatar: Integrate Gravatar login with WPJM

- Related to 105686-Automattic/gravatar
- Related to 105704-Automattic/gravatar

## Proposed Changes

* Make Gravatar login flow more generic by renaming the relevant components, functions, classes, etc.
* Integrate WP Job Manager OAuth2 client
* Align the magic link page with the design (eQKMA9DqH4V9rEfBTskC2q-fi)
* Show the link of the password/social login page for proxied A12s only
* Auto language switching based on the user's top priority locale

## PR Scope

The scope of this PR focuses on the front-end of the Gravatar-powered login flow.

## Testing Instructions



* [Setup Calypso](https://github.com/Automattic/wp-calypso#getting-started) and check out this PR
* Run `yarn start`
* Logout your account
* The magic login page of Gravatar and WPJM should look as below:

| Gravatar | WPJM |
| - | - |
| <img width="762" alt="截圖 2023-09-14 下午2 44 14" src="https://github.com/Automattic/wp-calypso/assets/21308003/6e15717a-06b6-46cd-8ef3-e93f1c3e36a6">| <img width="762" alt="截圖 2023-09-14 下午2 44 02" src="https://github.com/Automattic/wp-calypso/assets/21308003/b0cc7022-fb84-4beb-b007-94537c70edf9"> |

* The language of the above UIs will be automatically switched to the user's preferred locale (which is set by the Browser's language option)
* The link to the password/social login page won't be shown on production. Code review should be fine (the `configData.env_id ` will be: `development` / `staging` / `production`):

<img width="762" alt="截圖 2023-09-14 下午2 44 14" src="https://github.com/Automattic/wp-calypso/assets/21308003/58a75db2-04b5-4ac1-b5a3-8cb88378bb68">

* Click the signup link of the F2A pages and the password/social login page should go to the corresponding magic login page:

| F2A | Password/Social |
| - | - |
| <img width="761" alt="截圖 2023-09-14 下午2 58 41" src="https://github.com/Automattic/wp-calypso/assets/21308003/aca0eb7d-67f6-4b4a-9c6c-373c17b6a648"> | <img width="762" alt="截圖 2023-09-14 下午2 59 04" src="https://github.com/Automattic/wp-calypso/assets/21308003/d76ed1ef-72f3-41ec-a1da-d3c76b1ff647"> |

* Try to create a new account via the magic login page of [Gravatar](http://calypso.localhost:3000/log-in/link?client_id=1854&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fclient_id%3D1854%26response_type%3Dcode%26blog_id%3D0%26state%3D943e6fc69e6b8c3d5d488f7d9e3ee0083c5700a3136cc3f25205360ef89c825f%26redirect_uri%3Dhttps%253A%252F%252Fen.gravatar.com%252Fconnect%252F%253Faction%253Drequest_access_token%26from-calypso%3D1) and [WPJM](http://calypso.localhost:3000/log-in/link/?client_id=90057&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthenticate%2F%3Fclient_id%3D90057%26response_type%3Dcode%26state%3D01b7d7bfcfd126450053529ce09fefb5%26redirect_uri%3Dhttp%253A%252F%252Fwpjobmanagercom.vipdev.lndo.site%253A8000%252Fwp-json%252Fwpjmcom-auth%252Fv1%252Fcallback&create_account=1), it should work as expected
* Try to login via the magic login page of [Gravatar](http://calypso.localhost:3000/log-in/link?client_id=1854&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fclient_id%3D1854%26response_type%3Dcode%26blog_id%3D0%26state%3D943e6fc69e6b8c3d5d488f7d9e3ee0083c5700a3136cc3f25205360ef89c825f%26redirect_uri%3Dhttps%253A%252F%252Fen.gravatar.com%252Fconnect%252F%253Faction%253Drequest_access_token%26from-calypso%3D1) and [WPJM](http://calypso.localhost:3000/log-in/link/?client_id=90057&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthenticate%2F%3Fclient_id%3D90057%26response_type%3Dcode%26state%3D01b7d7bfcfd126450053529ce09fefb5%26redirect_uri%3Dhttp%253A%252F%252Fwpjobmanagercom.vipdev.lndo.site%253A8000%252Fwp-json%252Fwpjmcom-auth%252Fv1%252Fcallback&create_account=1), it should work as expected
* Try to login as an A11n via the password login page of [Gravatar](http://calypso.localhost:3000/log-in?redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fclient_id%3D1854%26response_type%3Dcode%26blog_id%3D0%26state%3D943e6fc69e6b8c3d5d488f7d9e3ee0083c5700a3136cc3f25205360ef89c825f%26redirect_uri%3Dhttps%253A%252F%252Fen.gravatar.com%252Fconnect%252F%253Faction%253Drequest_access_token%26from-calypso%3D1&client_id=1854) and [WPJM](http://calypso.localhost:3000/log-in?redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthenticate%2F%3Fclient_id%3D90057%26response_type%3Dcode%26state%3D01b7d7bfcfd126450053529ce09fefb5%26redirect_uri%3Dhttp%253A%252F%252Fwpjobmanagercom.vipdev.lndo.site%253A8000%252Fwp-json%252Fwpjmcom-auth%252Fv1%252Fcallback&client_id=90057), it should work as expected
* The [password login page](http://calypso.localhost:3000/log-in) and the [magic login page](http://calypso.localhost:3000/log-in/link) of WP.com should still work the same

## Pre-merge Checklist
